src/util-decode-mime.c

```diff
@@ -686,7 +686,11 @@ static uint8_t * GetFullValue(DataValue *dv, uint32_t *len)
 static inline uint8_t * FindBuffer(const uint8_t *src, uint32_t len, const uint8_t *find, uint32_t find_len)
 {
     /* Use utility search function */
-    return BasicSearchNocase(src, len, find, find_len);
+    if (find_len > UINT16_MAX) {
+        // TODO rewrite all this to have app-layer events
```

Comment:
ticket

Reply:
Thanks for pointing this
I did https://github.com/OISF/suricata/pull/7074
I managed to get rid of this check at this level (like `find_len` is always small enough) so I do not think we need a ticket for this